test_launch_ros/test/test_launch_ros/utilities/test_type_utils.py

```diff
@@ -0,0 +1,149 @@
+# Copyright 2020 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Test type checking/coercion utils."""
+
+from typing import List
+from typing import Union
+
+from launch_ros.utilities.type_utils import check_is_instance_of_valid_type
+from launch_ros.utilities.type_utils import check_type
+from launch_ros.utilities.type_utils import coerce_to_type
+
+import pytest
+
+
+def test_coercions_using_yaml_rules():
+    assert coerce_to_type('asd') == 'asd'
+    assert coerce_to_type('tRuE') == 'tRuE'
+    assert coerce_to_type("'1'") == '1'
+    assert coerce_to_type("'off'") == 'off'
+
+    assert coerce_to_type('1') == 1
+    assert coerce_to_type('1000') == 1000
+
+    assert coerce_to_type('1.') == 1.0
+    assert coerce_to_type('1000.0') == 1000.
+    assert coerce_to_type('0.2') == .2
+    assert coerce_to_type('.3') == 0.3
+
+    assert coerce_to_type('on') is True
+    assert coerce_to_type('off') is False
+    assert coerce_to_type('True') is True
+
+    assert coerce_to_type('[2, 1, 1]') == [2, 1, 1]
+    assert coerce_to_type('[.2, .1, .1]') == [.2, .1, .1]
+    assert coerce_to_type('[asd, bsd, csd]') == ['asd', 'bsd', 'csd']
+    assert coerce_to_type('[on, false, no]') == [True, False, False]
+
+
+def test_coercions_given_specific_type():
```

Comment:
@ivanpauno consider checking that coercion to a type different from the one that results of using `data_type=None` works e.g. ` assert coerce_to_type("1", data_type=str) == '1'`.